### PR TITLE
Add newrelic_security gem to changelog

### PR DESCRIPTION
**Pull request by kaylareopelle**
Adds a changelog entry for #2752

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,63 @@
...
+Version <dev> adds support for the `newrelic_security` agent, introduces instrumentation for the LogStasher gem, improves instrumentation for the `redis-clustering` gem, and updates the Elasticsearch instrumentation to only attempt to get the cluster name once per client, even if it fails.
+
+- **Feature: Add support for the newrelic_security agent**
+
+  [New Relic Interactive Application Security Testing (IAST)](https://docs.newrelic.com/docs/iast/introduction/) can help you prevent cyberattacks and breaches on your applications by probing your running code for exploitable vulnerabilities.
+
+  The `newrelic_security` gem provides this feature for Ruby. It depends on `newrelic_rpm`. This is the first version of `newrelic_rpm` compatible with `newrelic_security`.
+
+  At this time, the security agent is intended for use only within a dedicated security testing environment with data that can tolerate modification or deletion. The security agent is available as a separate Ruby gem, newrelic_security. It is recommended that this separate gem only be introduced to a security testing environment by leveraging Bundler grouping like so:
```
Comment (fallwith) [inline CHANGELOG.md:12]:
```suggestion
  At this time, the security agent is intended for use only within a dedicated security testing environment with data that can tolerate modification or deletion. The security agent is available as a separate Ruby gem, `newrelic_security`. It is recommended that this separate gem only be introduced to a security testing environment by leveraging Bundler grouping like so:
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,10 +1,63 @@
...
+    security:
+      <<: *default_settings
+      security.enabled: true
+      security.agent.enabled: true
+
+    production:
+      <<: *default_settings
+  ```
+
+  The following configuration relate to the `newrelic_security` gem:
```
Comment (kaylareopelle) [inline CHANGELOG.md:46]:
```suggestion
  The following configurations relate to the `newrelic_security` gem:
```

**Review by hannahramadan — APPROVED** [review]

**Comment by kaylareopelle** [comment]
@prateeksen - I drafted this entry for our changelog/release notes to describe the security agent release. Please review when you have a chance.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,10 +1,64 @@
...
+
+  At this time, the security agent is intended for use only within a dedicated security testing environment with data that can tolerate modification or deletion. The security agent is available as a separate Ruby gem, `newrelic_security`. It is recommended that this separate gem only be introduced to a security testing environment by leveraging Bundler grouping like so:
+
+  ```ruby
+    # Gemfile
+    gem 'newrelic_rpm'               # New Relic APM observability agent
+    gem 'newrelic-infinite_tracing'  # New Relic Infinite Tracing
+
+    group :security do
+      gem 'newrelic_security'        # New Relic security agent
```
Comment (prateeksen) [inline CHANGELOG.md:21]:
```suggestion
      gem 'newrelic_security', require: false        # New Relic security agent
```

Comment (prateeksen) [reply CHANGELOG.md:21]: @kaylareopelle `require: false` is mandatory to enforce loading of newrelic_security gem via `security.agent.enabled` flag.

**Review by prateeksen — APPROVED** [review]

**Review by tannalynn — APPROVED** [review]

**Pull request merged**